data/lm/generate_lm.py

```diff
@@ -4,74 +4,97 @@
 import os
 import subprocess
 from collections import Counter
+from itertools import zip_longest
+import shutil
 
 import progressbar
 
 
 def convert_and_filter_topk(args):
     """ Convert to lowercase, count word occurrences and save top-k words to a file """
 
-    counter = Counter()
-    data_lower = os.path.join(args.output_dir, "lower.txt.gz")
+    # Loop over multiple input_txt arguments
```

Comment:
To the best of my understanding, there does noi seems to be a huge difference within the body of the loop compared to the previous version, so I'm wondering if for readability we could keep `convert_and_filter_topk` a one-file level function that acts on one text file, and have a higher level function that  gets in charge of dumping everything.

The only drawback I can see for that is memory usage, so I'm wondering if maybe it is something you already experimented and came back from?

Reply:
@nmstoker I think this would be the biggest change in your PR, what's your take on that?

Reply:
Yes, I think this sounds like a good way forward.

Is it cleanest for me to do a new PR, handling the steps as distinct commits? (as I should've done first time!)

I was thinking I'd keep the keep `convert_and_filter_topk` a one-file level function, I'd try to avoid it passing the whole vocab in memory (as per the other comment), I'd split out the stats like you suggested and lastly I'd get it looping over the multiple files.

If okay, I'll get that sorted tomorrow or Sunday at the latest.

Reply:
> Is it cleanest for me to do a new PR, handling the steps as distinct commits? (as I should've done first time!)

No, just rebase / edit your history and force push?



> If okay, I'll get that sorted tomorrow or Sunday at the latest.

Sounds awesome!

Reply:
> No, just rebase / edit your history and force push?

Great, I'll do that 👍🏻